src/describe/sobjectApi.ts

```diff
@@ -0,0 +1,319 @@
+/*
+ * Copyright (c) 2020, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import { RequestInfo } from 'jsforce';
+import { Connection } from '../connection';
+import { Logger } from '../logger';
+
+/* tslint:disable:no-any */
+export interface SObject {
```

Comment:
We should move the types/interfaces/enums to a types specific file so they can be published to consumers of the library.